### PR TITLE
Update verify rules

### PR DESCRIPTION
**Pull request by tpfau**
Updating VerifyModel to:
1. Have better verbosity when testing model fields for verbosity
2. Explicity testing rule syntax (i.e. whether rules are actually evaluateable)

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [ ] with [X] to check the box)*

### CODE REVIEW

File: src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+%    ruleString:    The string in rule format (i.e. x(1) | x(2) & x(3) ...)
+%um
+% OUTPUT:
+%    tf:            Whether the given string is evaluateable (i.e. the
+%                   syntax is correct
+persistent x
+
+if isempty(x)
+    % 10000 genes should be large enough for most models.
+    x = true(10000,1);
```
Comment (laurentheirendt) [inline src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m]: Would you be able to base this number on the model size?

File: src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m [hunk 2]
```diff
@@ -0,0 +1,44 @@
+function tf = verifyRuleSyntax(ruleString)
+% try to verify if a given rule string is syntactically correct
```
Comment (laurentheirendt) [inline src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m]: This header needs proper formatting.

File: src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            x = true(max(vals),1);
+            temp = eval(ruleString);
+            % if we reach this point it is fine
+            tf = true;
+        catch ME
+            tf = false;
+        end
+    end
+end
+
```
Comment (laurentheirendt) [inline src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m]: There are many empty lines here.

File: src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    x = true(10000,1); 
+end
+
+if isempty(ruleString)
+    tf = true;
+else
+    try
+        temp = eval(ruleString);
+        % if we reach this point it is fine
+        tf = true;
```
Comment (laurentheirendt) [inline src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m]: `tf = true` could be set as default, no?

File: src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m (hunk 1 above)
Comment (tpfau) [reply src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m]: I would have to pass a model into the function, and the function as such does not need one. So I would rather not do so.

File: src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m (hunk 4 above)
Comment (tpfau) [reply src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m]: No, rather false.
But I just noticed, that this is still bugged. Need to fix the second try/catch block..

Comment (laurentheirendt) [reply src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m]: Yeah, doesn't matter 😄

**Comment by tpfau** [comment]
I'm not quite sure, what to do about the invalid rule in Recon2.04.mat.
There is a rule (2173) which has ' ... | ()' which is definitely invalid, and can't be evaluated, and this is also present in grRules, and thus can't be fixed by `generateRules`
@rmtfleming would it be possible to fix this rule in the Recon2.04.mat file on vmh, as thats the place this model is obtained from?

**Comment by tpfau** [comment]
Ok, I updated the convertOldStyle model function to explcicitly correct that one particular rule... This is ugly, but since there is no response, I don't see how else to make these updated work

**Comment by tpfau** [comment]
Lets retry this, as it works on my local machine and on different CI machines

**Comment by tpfau** [comment]
@laurentheirendt Any idea why the windows CI can't process the checkout?

**Comment by tpfau** [comment]
restarting again

**Comment by tpfau** [comment]
another restart. lets see if the win7 machine is broken or can recover.

**Comment by tpfau** [comment]
Any further changes needed or do I miss something?

**Comment by tpfau** [comment]
@laurentheirendt :
Can this be merged?

**Pull request merged**